hrpc/get.go

```diff
@@ -18,8 +19,41 @@ type Get struct {
 	baseQuery
 	// Don't return any KeyValue, just say whether the row key exists in the
 	// table or not.
-	existsOnly bool
-	skipbatch  bool
+	existsOnly  bool
+	skipbatch   bool
+	consistency QueryConsistency
+}
+
+// QueryConsistency is used to specify the required consistency of data
+//
+// See https://docs.cloudera.com/HDPDocuments/HDP2/HDP-2.2.9/bk_hadoop-ha/content/ha-hbase-timeline-consistency.html
+type QueryConsistency int
+
+const (
+	// Use HBase's default
+	DefaultConsistency QueryConsistency = iota
+
+	// Guarantees that the client receives the latest data.
+	StrongConsistency
+
+	// Client might receive stale data (indicated by the Stale field), but
+	// the data that is received was valid at a given point of time.
+	TimelineConsistency
+)
+
+func (c QueryConsistency) ToProto() (ret *pb.Consistency) {
```

Comment:
Can we make this a private symbol and avoid the use of a pointer to prevent memory allocations? See `ExistenceOnly` for an example.

Reply:
Made private.

The pointer _here_ is not the issue — the pointer in the protocol message is. There is no analogue of `proto.Bool` which `ExistenceOnly` uses. What we can do is have two global variables for each of the values of `pb.Consistency`. I think it's up to the maintainer if that clutter is worthwhile.